examples/ping/ping.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"flag"
+	"fmt"
+	"time"
+
+	"log"
+
+	gorpc "github.com/hsanjuan/go-libp2p-gorpc"
+	libp2p "github.com/libp2p/go-libp2p"
+	host "github.com/libp2p/go-libp2p-host"
+	peerstore "github.com/libp2p/go-libp2p-peerstore"
+	protocol "github.com/libp2p/go-libp2p-protocol"
+	multiaddr "github.com/multiformats/go-multiaddr"
+)
+
+type PingArgs struct {
+	Data []byte
+}
+type PingReply struct {
+	Data []byte
+}
+type PingService struct{}
+
+func (t *PingService) Ping(argType PingArgs, replyType *PingReply) error {
+	log.Println("Received a Ping call")
+	replyType.Data = argType.Data
+	return nil
+}
+
+func createPeer(listenAddr string) host.Host {
+	ctx, _ := context.WithCancel(context.Background())
```

Comment:
No benifit to creating with cancel, since the example doesn't make use of the cancel function.

Reply:
True, was shamelessly copied from some other libp2p example. Thanks